### PR TITLE
basicly support export to svg

### PR DESCRIPTION
**Pull request by Sean10**
But now only succeed to export need some steps.

1. must run view mind map.
2. manual adjust the zoom scale until could show completely.
3. run export svg.

Now you would get the svg file with the same name in the same directory.

 A important problem which still need help to deal with, and i'm not familiar with:
* adaptive scale

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
 		"Other"
 	],
 	"activationEvents": [
-		"onCommand:mdmmp.showMindMap"
+		"onCommand:mdmmp.showMindMap",
+		"onCommand:mdmmp.exportSVG"
```
Comment (baobao1270) [inline package.json]: I think using little camel-case is better, so can you change the command to `mdmmp.exportSvg`?

File: package.json [hunk 2]
```diff
@@ -71,5 +81,7 @@
 		"typescript": "^4.0.2",
 		"vscode-test": "^1.4.0"
 	},
-	"dependencies": {}
+	"dependencies": {
+		"html2svg": "^1.0.1"
```
Comment (baobao1270) [inline package.json]: You didn't import the html2svg library, this dependency is redundant. Please remove it.

File: src/mindMapPreviewWebview.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import * as vscode from 'vscode';
 import * as path from "path";
+import * as fs from "fs";
+// const html2svg = require('html2svg');
```
Comment (baobao1270) [inline src/mindMapPreviewWebview.ts]: Debug comments like this should not appear in the master branch. Please delete it. Thanks.

File: src/mindMapPreviewWebview.ts [hunk 4]
```diff
@@ -17,6 +23,7 @@ class MindMapPreview {
                 ]
             }
         );
+        console.log("path", path.join(context.extensionPath, 'html'));
```
Comment (baobao1270) [inline src/mindMapPreviewWebview.ts]: Debug logs like this should not appear in the master branch. Please delete it. Thanks.

File: src/mindMapPreviewWebview.ts [hunk 5]
```diff
@@ -47,12 +54,25 @@ class MindMapPreview {
...
-                <svg id="container"></svg>
+                <svg id="container" preserveAspectRatio="none meet"></svg>
                 <script>
                     const map = markmap.Markmap.create("#container", null, {"t":"heading","d":1,"p":{},"v":"Rendering, please wait..."});
+                    const vscode = acquireVsCodeApi();
                     window.addEventListener('message', event => {
                         const message = event.data;
-                        map.setData(markmap.transform(message));
+                        switch (message.command) {
+                            case "saveSVG": {
```
Comment (baobao1270) [inline src/mindMapPreviewWebview.ts]: There are 2 coding style problems to fix:
1. Please name the command by little camel-case, such as `saveSvg`
2. Please use the `<verb>-<object>` naming style, for example, `saveSvg`, `renderMarkdown`, `doSomething`.

**Review by baobao1270 — CHANGES_REQUESTED** [review]
I have tested the code of your code. A bug is found. Here is the way to reproduce:
1. Open a markdown file and run `mdmmp.showMindMap`
2. Run command `mdmmp.exportSVG`, and a SVG file is created.
3. Delete the SVG file. Then run command `mdmmp.exportSVG` again, but no SVG file is created.
4. Then close the preview webview and run `mdmmp.showMindMap` again, the extension crush.

I think your code make this extension unstable, so please test the bugs and fix it, then commit it again. Thank you for your contribution.

**Comment by Sean10** [comment]
Got it.

**Pull request merged**